radio/src/thirdparty/libopenui/src/mainwindow.cpp

```diff
@@ -43,6 +43,13 @@ void MainWindow::run(bool trash)
   auto opaque = Layer::getFirstOpaque();
   if (opaque) opaque->checkEvents();
 
+  auto copy = children;
+  for (auto child: copy) {
+    if (!child->deleted() && child->isBubblePopup()) {
```

Comment:
You might want to have a look at LVGL timers instead of this. You could just call a timer that would trigger timeout handling.

Reply:
Doing it this way allows for a future change to remove the _run_popup_dialog message loop and use the normal message loop for handling POPUP_WARNING and POPUP_INFORMATION dialogs.
It might also work for the FullScreeDialog popup.

Reply:
I see, wouldn't be bad. Ideally we should however move to some mechanism that does not require to poll `checkEvents()`. But maybe I'm just dreaming too much 🥹